### DISCUSSION TITLE
ImageSharp LanczosResampler not working anymore

### DISCUSSION THREAD

**Discussion by feliwir** [SixLabors/ImageSharp#1282] [Q&A] [answered]
Hello everyone,

after updating to rc0003 (from beta0006) the overload of the Resize function that takes a custom IResampler crashes with an LanczosResampler. The line of code is as follows:
```csharp
...
tgaImage.Mutate(x => x.Resize(384, 384, new LanczosResampler()));
...
```

Exception:
```
SixLabors.ImageSharp.ImageProcessingException: "An error occurred when processing the image using ResizeProcessor`1. See the inner exception for more detail."
```

Inner exception:
```
ArgumentOutOfRangeException: Specified argument was out of the range of valid values.
```

Callstack:
```
    System.ThrowHelper.ThrowArgumentOutOfRangeException()
    System.Span<T>.Slice(int, int)
    SixLabors.ImageSharp.Memory.Buffer2D<T>.GetRowSpan(int)
    SixLabors.ImageSharp.Processing.Processors.Transforms.ResizeKernelMap.CreateKernel(int, int, int)
    SixLabors.ImageSharp.Processing.Processors.Transforms.ResizeKernelMap.BuildKernel<TResampler>(TResampler, int, int)
    SixLabors.ImageSharp.Processing.Processors.Transforms.ResizeKernelMap.PeriodicKernelMap.Initialize<TResampler>(TResampler)
    SixLabors.ImageSharp.Processing.Processors.Transforms.ResizeKernelMap.Calculate<TResampler>(TResampler, int, int, SixLabors.ImageSharp.Memory.MemoryAllocator)
    SixLabors.ImageSharp.Processing.Processors.Transforms.ResizeProcessor<TPixel>.ApplyTransform<TResampler>(TResampler)
    SixLabors.ImageSharp.Processing.Processors.Transforms.ResizeProcessor<TPixel>.SixLabors.ImageSharp.Processing.Processors.Transforms.IResamplingTransformImageProcessor<TPixel>.ApplyTransform<TResampler>(TResampler)
    SixLabors.ImageSharp.Processing.Processors.Transforms.LanczosResampler.ApplyTransform<TPixel>(SixLabors.ImageSharp.Processing.Processors.Transforms.IResamplingTransformImageProcessor<TPixel>)
```

The exact code used to work before, so i guess that **LanczosResampler** got some changes maybe?

**Comment by JimBobSquarePants** [comment, marked as answer]
It’s a struct now. Use `KnownResamplers` https://docs.sixlabors.com/api/ImageSharp/SixLabors.ImageSharp.Processing.KnownResamplers.html

**Reply by feliwir** [reply]
Thanks found it aswell now :)

**Reply by feliwir** [reply]
@JimBobSquarePants what was the default smoothing radius in earlier ImageSharp versions?

**Reply by JimBobSquarePants** [reply]
Not sure I understand the question. Default resampler has always been Bicubic

**Reply by feliwir** [reply]
We used to use just `new LanczosResampler()` and now i'm wondering what the default smoothing radius was for that (Lanczos2, Lanczos3 etc.). So pretty much which of the new KnownResamplers mimics the old **new LanczosResampler()**.

Sorry for formulating the question unclear :)

**Reply by JimBobSquarePants** [reply]
I’d have to check the old binary. Likely 3 though

**Reply by JimBobSquarePants** [reply]
Are you sure it was `beta0006`? There's no sign of that class in the assembly, tbh I don't ever remember it. 

![image](https://user-images.githubusercontent.com/385879/87873342-b79e6d00-c9b8-11ea-873e-d4713e33c207.png)

**Reply by feliwir** [reply]
We were using *Lanczos2Resampler*, which also answers my question about the radius we had. 
Sorry for the inconvenience, i was very confused

**Reply by JimBobSquarePants** [reply]
Haha... no worries, Glad you figured it out.

**Comment by feliwir** [comment]
Okay, fixed this myself already.
For everyone getting the same issues after upgrading:
Use **LanczosResampler.Lanczos3** (or some other static member depending on your needs)
